src/components/TaskCard/AddTaskCard.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { v1 as uuidv1 } from "uuid";
+import { sanitizeText } from "utils/helpers/sanitizeText.js";
+import PropTypes from "prop-types";
+import { addTask } from "store/actions/";
+import style from "components/TaskCard/index.module.scss";
+import { ENTER } from "utils/constant";
+const AddTaskCard = ({ showCreateCard, setShowCreateCard }) => {
+  const [inputText, setInputText] = useState("");
+  const [error, setError] = useState(null);
+  const dispatch = useDispatch();
+
+  const handleInputText = (e) => {
+    setInputText(e.target.value);
+  };
+
+  const storeTask = () => {
+    const task = sanitizeText(inputText);
+
+    if (task === "") {
+      setError("Please add task description");
+
+      return;
+    }
+
+    setShowCreateCard(!showCreateCard);
+
+    dispatch(
+      addTask({
+        id: uuidv1(),
+        task,
+        createdTime: new Date(),
+        completed: false,
+      })
```

Comment:
action `addTask` only need to pass the task name from the view. rest of the data can be set inside the reducer.

```suggestion
      addTask({ task })
```